File: C++/Pangram.cpp
#include <string>
#include <vector>
#include <numeric>
#include <iostream>
#include <algorithm>

std::string isPangram(const std::string& word) 
{
    std::vector<int> letter_counts(26, 0);
    for (char letter : word) 
    {
        int index = letter - 'a';
        letter_counts[index]++;
    }
    bool is_nonperfect_pangram = std::all_of(letter_counts.begin(), letter_counts.end(), [](int count){ return count > 0; });
    if (!is_nonperfect_pangram)
        return "NIE";
    int first_count = -1;
    for(int count : letter_counts)
    {
        if(count > 0)
        {
            first_count = count;
            break;
        }
    }
    bool is_perfect_pangram = std::all_of(letter_counts.begin(), letter_counts.end(), [first_count](int count){ return count == first_count; });
    if (is_perfect_pangram)
        return "PANGRAM PERFEKCYJNY";
    else
        return "PANGRAM";
}

int main() 
{
    std::string word;
    std::cin >> word;
    std::cout << isPangram(word);
    return 0;
}

File: Python/Pangram.py
def is_pangram(word):
    letter_counts = [0] * 26
    for letter in word:
        index = ord(letter) - ord('a')
        letter_counts[index] += 1
    is_nonperfect_pangram = all(count > 0 for count in letter_counts)
    if not is_nonperfect_pangram:
        return "NIE"
    first_count = next((x for x in letter_counts if x > 0), None)
    if all(count == first_count for count in letter_counts):
        return "PANGRAM PERFEKCYJNY"
    else:
        return "PANGRAM"

word = input()
print(is_pangram(word))